addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationAPI.java

```diff
@@ -0,0 +1,148 @@
+        this.addApiView(new ApiView(VIEW_PLAN_PROGRESS, new String[] {PARAM_PLAN_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        LOG.debug("handleApiAction {} {}", name, params);
+
+        if (name.equals(ACTION_RUN_PLAN)) {
+            try {
+                AutomationPlan plan =
+                        extension.loadPlan(
+                                new File(ApiUtils.getNonEmptyStringParam(params, PARAM_FILE_PATH)));
+                extension.registerPlan(plan);
+
+                if (View.isInitialised()) {
+                    extension.displayPlan(plan);
+                }
+                extension.runPlanAsync(plan);
+
+                return new ApiResponseElement(PARAM_PLAN_ID, Integer.toString(plan.getId()));
+            } catch (IOException | ApiException e) {
+                throw new ApiException(Type.DOES_NOT_EXIST, e.getMessage());
+            }
+        } else {
+            throw new ApiException(Type.BAD_ACTION);
+        }
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        LOG.debug("handleApiView {} {}", name, params);
+
+        if (name.equals(VIEW_PLAN_PROGRESS)) {
+            AutomationPlan plan = extension.getPlan(params.getInt(PARAM_PLAN_ID));
+
+            if (plan == null) {
+                throw new ApiException(Type.DOES_NOT_EXIST);
+            }
+
+            return progressResponse(plan);
+        } else {
+            throw new ApiException(Type.BAD_VIEW);
+        }
+    }
+
+    ApiResponse progressResponse(AutomationPlan plan) {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put(PARAM_PLAN_ID, plan.getId());
+
+        if (plan.getStarted() != null) {
+            map.put("started", plan.getStarted());
+        }
+        if (plan.getFinished() != null) {
+            map.put("finished", plan.getFinished());
+        }
+
+        ApiResponseList infoMsgs = new ApiResponseList(ELEMENT_INFO);
+        plan.getProgress().getInfos().stream()
+                .map(msg -> new ApiResponseElement(ELEMENT_INFO, msg))
+                .forEach(infoMsgs::addItem);
+        map.put(ELEMENT_INFO, infoMsgs);
+
+        ApiResponseList warnMsgs = new ApiResponseList(ELEMENT_WARN);
+        plan.getProgress().getWarnings().stream()
+                .map(msg -> new ApiResponseElement(ELEMENT_WARN, msg))
+                .forEach(warnMsgs::addItem);
+        map.put(ELEMENT_WARN, warnMsgs);
+
+        ApiResponseList errorMsgs = new ApiResponseList(ELEMENT_ERROR);
+        plan.getProgress().getErrors().stream()
+                .map(msg -> new ApiResponseElement(ELEMENT_ERROR, msg))
+                .forEach(errorMsgs::addItem);
+        map.put(ELEMENT_ERROR, errorMsgs);
+
+        return new ApiResponseSet<>("progress", map);
```

Comment:
Ideally it would be called `VIEW_PLAN_PROGRESS`, that's the normal/expected name.